### DISCUSSION TITLE
How to filter in firestore query

### DISCUSSION THREAD

**Discussion by anhnch** [invertase/react-native-firebase#6704] [Q&A]
Hi, I have a collection of realestate posts in Firestore. Every post has bunch of fields: number of bed rooms, number of restrooms, gross area, project, completion status, door direction, balcony direction, for sale/lease, price, floor, code, address, type of legal paper, with or without interiors and many more.

I have a filter button which allow users to enter the number of bedrooms, floor, price range... any of the realestate post's fields then filter the list. This is very simple in SQL query, but I'm not sure what can I do with Firestore. 

Should I create indices for any combination of realestate post fields that the user may filter?

**Comment by mikehardy** [comment]
Hi there! General support for firebase / firestore is on stackoverflow, where you will frequently get answers from the upstream google firebase team members. If you are just starting out with firestore, I highly recommend the firebase team's youtube channel, they have series of videos on each of the firebase technologies including firestore of course, and there are also codelabs where you can play around with the queries to see how things work.

**Comment by anhnch** [comment]
This link explains taking advantages of merging indices, but it doesn't seem to work
https://firebase.google.com/docs/firestore/query-data/index-overview

A few people having the same issue
https://stackoverflow.com/questions/65246473/how-to-get-firestore-index-merge-to-work
- They said Firestore still requires creating composite indices rather than merging. I confirmed this.
- They also said merging index is slower. I'm not sure for this.

So I ended up with
1. Just make the query. If the combination has not been indexed
2. Click the generated link in the console to create

And I got 30 indices, that's crazy!

**Reply by mikehardy** [reply]
Is this something that is actionable in react-native-firebase, or is this a general issue with firestore and firestore-android-sdk / firestore-ios-sdk?

For it to be actionable here, there would have to be some deficiency in the react-native-firebase implementations correctable by a code change in this module. If that's the case, could you point out exactly what the deficiency in react-native-firebase is and what sort of change here in this repo would make sense?

**Reply by anhnch** [reply]
It's not react-native-firebase related indeed. It's a general issue of firestore. That's why I post in Q&A discussion instead of issue section, to get advice from people who has experienced this. Anyway thank you for your reply. I'll post it to stackoverflow instead.
